pilot/pkg/model/push_context.go

```diff
@@ -2167,9 +2167,13 @@ func (ps *PushContext) ServiceAccounts(hostname host.Name, namespace string, por
 func (ps *PushContext) SupportsTunnel(ip string) bool {
 	infos, _ := ps.ambientIndex.PodInformation(sets.New(types.NamespacedName{Name: ip}))
 	for _, p := range infos {
-		if p.Protocol == workloadapi.Protocol_HTTP {
-			return true
+		switch addr := p.Address.Type.(type) {
+		case *workloadapi.Address_Workload:
```

Comment:
how do we know if a Service supports ztunnel? The spec doesn't include protocol - so wouldn't it be dependent on the individual workloads/endpoints for that Service?

Reply:
ztunnel supports all services? we still need services for VIP -> workload mapping, regardless of HBONE/"ztunnel support" (what exactly do you mean by this?)

ztunnel can resolve any svc VIP to workload IPs and then apply policy (e.g. rbac) to it

the only services ztunnel can't make sense of are the ones that don't have a workload that maps to the service VIP

Reply:
I don't think Service can support tunnel - it is a property of the workload. 

However it may make sense to treat some services and ServiceEntry in a 'special' way - for example use a 'via' waypoint, skip load balancing and use the service VIP directly. 

But separate discussion and PR...

Reply:
after latest design discussion, yes, Service should NOT support tunnel and is only referring to Workloads